### ISSUE TITLE
Illegal (or unknown) operand: What does this mean?

### ISSUE THREAD

**Issue by thereAreDemonsNearby** [ithemal/Ithemal#60]
Illegal (or unknown) operand:
$0x000055e40e799337
In instruction:
jle    $0x000055e40e799337

**Comment by alexrenda** [comment]
Not entirely sure why the reported error is that the operand is unknown, but Ithemal only operates over basic blocks so it won't be able to generate a prediction for a jump instruction

**Comment by andreas-abel** [comment]
But a jump instruction can be part of a basic block if it is the last instruction in this block.

**Comment by alexrenda** [comment]
That's true, and that is a limitation of Ithemal's current model — different jump instructions could plausibly result in different timings. Unfortunately, including jump instructions at the end of basic blocks makes it quite a bit harder (though potentially not impossible) to get an accurate measure of ground-truth throughput, so we don't include that currently. It would definitely be interesting to see examples of blocks where the choice of jump instruction has a real impact on throughput though.

**Comment by thereAreDemonsNearby** [comment]
Thank you for your reply. It's true that there are several if statements in the loop body I've tested using Ithemal.  It seems that IACA or llvm-mca is more suitable for such code snippets
